updater.sh

```diff
@@ -1,22 +1,70 @@
 #!/bin/bash
 
 cat ~/.logo
-printf "\n\n\n"
 
-RED=$(tput setaf 1)
 GREEN=$(tput setaf 2)
+RED=$(tput setaf 1)
 NORMAL=$(tput sgr0)
 
+PKG=""
+
 printProgress() {
-    if [[ "$2" == "starting" ]]; then
-	    printf "${RED}\n$1 $2\n${NORMAL}"
+    if [[ "$1" == *starting ]] || [[ "$1" == *privileges ]] || [[ "$1" == *aborting* ]]; then
+	    printf "${RED}\n$1\n${NORMAL}"
     else
-	    printf "${GREEN}$1 $2\n${NORMAL}"
+	    printf "${GREEN}$1\n${NORMAL}"
     fi
 }
 
-printf "${RED}This script requires sudo to run commands.\n${NORMAL}"
-PKG=""
+if [[ -n "$(command -v pkg)" ]]; then
+    PKG="pkg"
+
+    printf "${GREEN}System detected: ${RED}Using $PKG\n${NORMAL}"
+
+    printProgress "update: starting"
+    $PKG update
+    printProgress "update: completed"
+
+    printProgress "upgrade: starting"
+    $PKG upgrade
+    printProgress "upgrade: completed"
+
+    printProgress "autoclean: starting"
+    $PKG autoclean
+    printProgress "autoclean: completed\n"
+
+    exit 1;
+elif [[ -n "$(command -v emerge)" ]]; then
+    PKG="emerge"
+
+    printf "${GREEN}System detected: ${RED}Using $PKG\n${NORMAL}"
+
+    printProgress "syncing: starting"
+    $PKG --sync
+    printProgress "syncing: completed"
+
+    printProgress "update: starting"
+    $PKG --update --deep --newuse --with-bdeps y @world --ask
+    printProgress "update: completed"
+
+    printProgress "deepclean: starting"
+    $PKG --depclean --ask
+    revdep-rebuild
+    printProgress "deepclean: completed\n"
+
+    exit 1;
+fi
+
+printProgress "Checking for sudo privileges"
+sudo -v >/dev/null 2>&1
+if [[ "$(echo $?)" -eq 0 ]]; then
+    printProgress "Access granted.\n"
+    SUDO="sudo"
```

Comment:
This variable isn't used anymore after the definition. @archetipico

Reply:
I forgot to remove it 😮